### PR TITLE
Calico 3.6

### PR DESCRIPTION
**Pull request by jnummelin**
I did a bit of refactoring on Calico manifests to make it easier to keep in sync with their official manifests:
- All CRDs combined into one single yaml file
- RBACs divided into two files, one for calico-node and one for calico-controller
  - Each file has all RBAC related things, roles, SA and binding
- CNI install moved to be an `initContainer` as it is in "official" manifests
- Adds new deployment for calico-controller

TODOs:

- [x] Use correct CDN for images

### CODE REVIEW

File: lib/pharos/resources/calico/20-configmap.yml.erb [hunk 1]
```diff
@@ -7,6 +7,15 @@ metadata:
...
   ipv4_pool_cidr: "<%= ipv4_pool_cidr %>"
 
+
+
+  # Configure the Calico backend to use.
+  calico_backend: "bird"
+
+  # TODO Do we need mtu setting?
+  # Configure the MTU to use
+  veth_mtu: "1440"
```
Comment (jakolehm) [inline lib/pharos/resources/calico/20-configmap.yml.erb]: How is this mtu different vs cni_network_config mtu?

Comment (jnummelin) [reply lib/pharos/resources/calico/20-configmap.yml.erb]: By accident, good catch :) Fixed now.

Comment (jakolehm) [reply lib/pharos/resources/calico/20-configmap.yml.erb]: Ok, but why do we need both?

Comment (jakolehm) [reply lib/pharos/resources/calico/20-configmap.yml.erb]: > When using Kubernetes self-hosted manifests, the CNI plugin derives the MTU value from the veth_mtu field of the calico-config ConfigMap, and it is set to 1440 by default. On restart of the calico/node pods, any references to `__CNI_MTU__` are replaced by the veth_mtu value and inserted into the CNI configuration file (aka conflist) at the directory specified by Kubernetes (currently defaults to /etc/cni/net.d/).

Comment (jnummelin) [reply lib/pharos/resources/calico/20-configmap.yml.erb]: Their reference manifests have it in both places so didn't really dare to remove.

File: lib/pharos/resources/calico/20-configmap.yml.erb [hunk 2]
```diff
@@ -7,14 +7,11 @@ metadata:
...
 
-
-
   # Configure the Calico backend to use.
   calico_backend: "bird"
 
-  # TODO Do we need mtu setting?
   # Configure the MTU to use
-  veth_mtu: "1440"
+  veth_mtu: "1500"
```
Comment (jakolehm) [inline lib/pharos/resources/calico/20-configmap.yml.erb]: Not sure if this is the correct number for all modes, we use ipip by default which most likely should use `1440` ..

File: lib/pharos/config_schema.rb [hunk 3]
```diff
@@ -124,6 +124,7 @@ def self.messages
             optional(:ipip_mode).filled(included_in?: %w(Always CrossSubnet Never))
             optional(:nat_outgoing).filled(:bool?)
             optional(:environment).filled(:hash?)
+            optional(:mtu).filled(:int?, gt?: 0)
```
Comment (jakolehm) [inline lib/pharos/config_schema.rb]: Requires docs pr.

Comment (jnummelin) [reply lib/pharos/config_schema.rb]: Docs PR at https://github.com/kontena/pharos-docs/pull/172

**Pull request merged**